### DISCUSSION TITLE
New module: fetch-kotlinx-serialization for typed responses

### DISCUSSION THREAD

**Discussion by okarmazin** [varabyte/kobweb#599] [Ideas]
It might be beneficial to provide a fist-party module that adds convenience extensions on `HttpFetcher` for deserializing responses to typed values using `kotlinx.serialization`.

`window.http.get<MyType>(...)`, or `window.http.get(MyType.serializer(), ...)` if the reified lookup mechanism isn't available in Kotlin/Js.

**Comment by bitspittle** [comment]
We actually have some precedence for this, with https://github.com/varabyte/kobweb/tree/main/common/kobwebx-serialization-kotlinx used to generate Kotlinx serialization support for Kobweb workers (which require users register serialization logic to convert between app and worker)

I'd be tempted to say we can probably even put such helper methods inside that module without needing to create a new one.

**Reply by bitspittle** [reply]
Opened https://github.com/varabyte/kobweb/issues/602
